dds/DCPS/RTPS/ICE/Ice.h

```diff
@@ -225,6 +226,206 @@ class OpenDDS_Rtps_Export Agent {
   static Agent* instance();
 };
 
+class ServerReflexiveStateMachine {
+public:
+  enum StateChange {
+    SRSM_None,
+    SRSM_Set,
+    SRSM_Unset,
+    SRSM_Change
+  };
+
+  ServerReflexiveStateMachine()
+    : indication_count_(0)
+  {}
+
+  // Return Unset if transitioning from a determined SRA to an undetermined SRA.
+  // Return None otherwise.
+  StateChange send(const ACE_INET_Addr& address, size_t indication_count_limit, const DCPS::GuidPrefix_t& guid_prefix)
```

Comment:
A few of these methods seem to be too large to be defined inline (in the class definition).